baremaps-ogcapi/src/main/java/com/baremaps/ogcapi/BaseUriRequestFilter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2020 The Baremaps Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.baremaps.ogcapi;
+
+import java.io.IOException;
+import javax.inject.Singleton;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.core.*;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+@PreMatching
+@Singleton
```

Comment:
Do these annotations make the registering of the filter unnecessary? If so, we could use the same method for other filter.

Reply:
Yes, the @Provider annotation makes the class discoverable if we tell the container to scan the package.